d3a_interface/sim_results/all_results.py

```diff
@@ -60,6 +62,14 @@ def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str):
             v.update(area_dict, core_stats, current_market_slot)
         self._update_memory_utilization()
 
+    def update_from_repr(self, area_representation: Dict):
+        """
+        Can be added as an abstract method to the results_abc if needed
+        """
+        for key, value in self.results_mapping.items():
```

Comment:
Still we do not need the key here, how about:
```suggestion
        for result_object in self.results_mapping.values():
```
Also would be great if you did the same refactoring for the `update()` method right above